lib/validators/index.js

```diff
@@ -0,0 +1,252 @@
+/*!
+ * Copyright (c) 2022 Digital Bazaar, Inc. All rights reserved.
+ */
+
+import * as didErrors from './didErrors.js';
+import * as didKeyErrors from './didKeyErrors.js';
+import * as didResolverErrors from './didResolverErrors.js';
+import {IdDecoder} from 'bnid';
+
+const idDecoder = new IdDecoder();
+// multicodec ed25519-pub header as Uint8Array
+const MULTICODEC_ED25519_PUB_HEADER = new Uint8Array([0xed, 0x01]);
+
+/**
+ * Validates the components of a did.
+ *
+ * @param {object} options - Options to use.
+ * @param {object} options.didKeyComponents - An object with
+ *   various did components.
+ * @param {Array<string>} options.supportedMethods - A list of
+ *   did:key methods this implementation supports.
+ *
+ * @throws {Error} Throws did related errors.
+ *
+ * @returns {boolean} - Are the did components valid?
+ */
+// fixme this should probably be moved into the did-method-key
+// get function and be run before returning a didDocument
+export const validateDidRequest = ({
+  didKeyComponents,
+  supportedMethods,
+}) => {
+  const {scheme, method, version, multibase, did, isDidUrl} = didKeyComponents;
+  // this is for conformance with the did resolver spec
+  if(!supportedMethods.includes(method)) {
+    throw new didResolverErrors.MethodNotSupported(
+      `Unsupported method ${method}`);
+  }
+  _validateDid({scheme, did, isDidUrl});
+  if((isDidUrl && !_isValidDidUrl({didUrl: did}))) {
+    throw new didErrors.InvalidDidUrl(`Invalid didUrl ${did}`);
+  }
+  _validateDidKey({method, version, multibase});
+  _validateMultibaseEd25519({didKeyComponents});
```

Comment:
These should be broken out into other libraries -- what happens if you don't feed a did:key into this method?

Reply:
Hence issues like this: https://github.com/digitalbazaar/ed25519-verification-key-2020/issues/18

The problem is I couldn't get consensus in any realistic time frame to add these errors to this libraries. So the validators are created in really small functions that can be ported over to other libraries when consensus is achieved.

Reply:
There validators have been moved to their respective libraries. See https://github.com/digitalbazaar/did-method-key/pull/47